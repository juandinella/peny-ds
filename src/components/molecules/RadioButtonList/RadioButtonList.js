import React from 'react'
import PropTypes from 'prop-types'
import RadioButton from '../../atoms/RadioButton'
import styles from './RadioButtonList.styles'

/**
 * List of radio buttons.
 */

const RadioButtonList = ({
  id,
  radioList,
  radioName,
  className,
  horizontal,
  disabled,
  shouldRender,
}) => {
  return (
    <div
      css={styles}
      id={id}
      data-testid={id}
      className={className}
      horizontal={horizontal}
    >
      {radioList.map(({ checked, value, label, id, onChange }) => (
        <RadioButton
          key={id}
          id={id}
          radioName={radioName}
          checked={checked}
          label={label}
          value={value}
          onChange={disabled ? null : onChange}
          shouldRender={shouldRender}
          disabled={disabled}
        />
      ))}
    </div>
  )
}

RadioButtonList.displayName = 'RadioButtonList'

RadioButtonList.propTypes = {
  /** RadioButtonList's array */
  radioList: PropTypes.array,
  /** RadioButtonList's name */
  radioName: PropTypes.string.isRequired,
  /** RadioButtonList's id */
  id: PropTypes.string,
  /** To add style */
  className: PropTypes.string,
  /** Disabled RadioButton */
  disabled: PropTypes.bool,
  /** Horizontal layout (justify-content: space-between) */
  horizontal: PropTypes.bool,
  /**
   * Function that decides if the component should be rendered.
   *
   * @return {boolean} false if component should not be rendered and show skeleton instead. Component renders otherwise.
   */
  shouldRender: PropTypes.func,
}

RadioButtonList.defaultProps = {
  checked: false,
}

export default RadioButtonList
