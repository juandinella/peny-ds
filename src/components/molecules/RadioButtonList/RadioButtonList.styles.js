import { css } from 'styled-components'

export default css`
  width: 100%;

  > div {
    margin-bottom: var(--spacing-xxxs);
  }

  ${(props) =>
    props.horizontal &&
    css`
      display: flex;
      justify-content: space-between;
      width: 100%;
    `}
`
