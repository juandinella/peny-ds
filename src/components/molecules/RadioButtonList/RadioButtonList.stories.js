import RadioButtonList from './RadioButtonList'
import { getTemplate } from '../../../helpers/storybook'

const Template = getTemplate(RadioButtonList)

const radioOne = {
  label: 'Radio One',
  value: 'one',
  checked: true,
  id: 'radio-one',
}

const radioTwo = {
  label: 'Radio Two',
  value: 'two',
  checked: false,
  id: 'radio-two',
}

export default {
  title: 'Molecules/RadioButtonList',
  component: RadioButtonList,
  args: {
    radioList: [radioOne, radioTwo],
    radioName: 'radio-name',
  },
  argTypes: {},
}

export const Default = Template.bind({})

export const Horizontal = Template.bind({})
Horizontal.args = { horizontal: true }

export const Skeleton = Template.bind({})
Skeleton.args = { shouldRender: () => false }
