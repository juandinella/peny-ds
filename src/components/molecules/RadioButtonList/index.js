import RadioButtonList from './RadioButtonList'

export default RadioButtonList
