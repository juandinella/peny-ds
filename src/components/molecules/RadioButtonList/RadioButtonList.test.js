import React from 'react'

import { render, cleanup } from '@testing-library/react'

import RadioButtonList from './RadioButtonList'

import initStoryshots from '@storybook/addon-storyshots'

initStoryshots({
  storyKindRegex: /^Molecules\/RadioButtonList$/,
})

const radioOne = {
  label: 'Radio One',
  value: 'one',
  checked: true,
  id: 'radio-one',
}

const radioTwo = {
  label: 'Radio Two',
  value: 'two',
  checked: false,
  id: 'radio-two',
}

const componentProps = {
  id: 'unique_id',
  radioName: 'radio-name',
  className: 'custom_class',
  radioList: [radioOne, radioTwo],
}

afterEach(cleanup)

describe('<RadioButtonList />', () => {
  test('should render component correctly', () => {
    const { queryByTestId } = render(<RadioButtonList {...componentProps} />)

    const radiolist = queryByTestId(componentProps.id)

    expect(radiolist).toBeTruthy()
  })

  test('should render skeleton', () => {
    const { queryByTestId } = render(
      <RadioButtonList {...componentProps} shouldRender={() => false} />
    )

    expect(queryByTestId(`${radioOne.id}_skeleton`)).toBeTruthy()
    expect(queryByTestId(`${radioTwo.id}_skeleton`)).toBeTruthy()
  })

  test('should render class correctly', () => {
    const { queryByTestId, rerender } = render(
      <RadioButtonList {...componentProps} />
    )

    const radioList = queryByTestId(componentProps.id)

    expect(radioList.className).toContain('custom_class')

    rerender(<RadioButtonList {...componentProps} className={null} />)

    expect(radioList.className).not.toContain('custom_class')
  })
})
