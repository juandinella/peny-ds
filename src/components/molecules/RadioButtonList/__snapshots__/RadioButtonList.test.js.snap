// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots Molecules/RadioButtonList Default 1`] = `
<div
  style={
    Object {
      "alignContent": "flex-start",
      "display": "flex",
      "flexDirection": "column",
      "flexWrap": "wrap",
      "gap": "10px 30px",
      "height": "100%",
      "justifyContent": "flex-start",
      "maxHeight": "auto",
    }
  }
>
  <div
    className="sc-1gtlsc7-0 cDJPBe"
  >
    <div
      className="sc-6kv2s7-0 jTNQcl"
      data-testid="radio-one"
    >
      <input
        defaultChecked={true}
        id="radio-one"
        name="radio-name"
        type="radio"
        value="one"
      />
      <label
        htmlFor="radio-one"
      >
        Radio One
      </label>
    </div>
    <div
      className="sc-6kv2s7-0 jTNQcl"
      data-testid="radio-two"
    >
      <input
        defaultChecked={false}
        id="radio-two"
        name="radio-name"
        type="radio"
        value="two"
      />
      <label
        htmlFor="radio-two"
      >
        Radio Two
      </label>
    </div>
  </div>
</div>
`;

exports[`Storyshots Molecules/RadioButtonList Horizontal 1`] = `
<div
  style={
    Object {
      "alignContent": "flex-start",
      "display": "flex",
      "flexDirection": "column",
      "flexWrap": "wrap",
      "gap": "10px 30px",
      "height": "100%",
      "justifyContent": "flex-start",
      "maxHeight": "auto",
    }
  }
>
  <div
    className="sc-1gtlsc7-0 lcNqZR"
  >
    <div
      className="sc-6kv2s7-0 jTNQcl"
      data-testid="radio-one"
    >
      <input
        defaultChecked={true}
        id="radio-one"
        name="radio-name"
        type="radio"
        value="one"
      />
      <label
        htmlFor="radio-one"
      >
        Radio One
      </label>
    </div>
    <div
      className="sc-6kv2s7-0 jTNQcl"
      data-testid="radio-two"
    >
      <input
        defaultChecked={false}
        id="radio-two"
        name="radio-name"
        type="radio"
        value="two"
      />
      <label
        htmlFor="radio-two"
      >
        Radio Two
      </label>
    </div>
  </div>
</div>
`;

exports[`Storyshots Molecules/RadioButtonList Skeleton 1`] = `
<div
  style={
    Object {
      "alignContent": "flex-start",
      "display": "flex",
      "flexDirection": "column",
      "flexWrap": "wrap",
      "gap": "10px 30px",
      "height": "100%",
      "justifyContent": "flex-start",
      "maxHeight": "auto",
    }
  }
>
  <div
    className="sc-1gtlsc7-0 cDJPBe"
  >
    <div
      className="sc-kjnynt-0 jfZhXD"
      data-testid="radio-one_skeleton"
      id="radio-one_skeleton"
      style={
        Object {
          "height": undefined,
          "width": undefined,
        }
      }
    >
      <div
        className="sc-6kv2s7-0 jTNQcl"
        data-testid="radio-one"
      >
        <input
          defaultChecked={true}
          id="radio-one"
          name="radio-name"
          type="radio"
          value="one"
        />
        <label
          htmlFor="radio-one"
        >
          Radio One
        </label>
      </div>
    </div>
    <div
      className="sc-kjnynt-0 jfZhXD"
      data-testid="radio-two_skeleton"
      id="radio-two_skeleton"
      style={
        Object {
          "height": undefined,
          "width": undefined,
        }
      }
    >
      <div
        className="sc-6kv2s7-0 jTNQcl"
        data-testid="radio-two"
      >
        <input
          defaultChecked={false}
          id="radio-two"
          name="radio-name"
          type="radio"
          value="two"
        />
        <label
          htmlFor="radio-two"
        >
          Radio Two
        </label>
      </div>
    </div>
  </div>
</div>
`;
